src/controllers/credentials.ts

```diff
@@ -101,6 +103,13 @@ export class CredentialController {
 			request.body['@context'] = [request.body['@context']];
 		}
 
+		const resolvedResult = await new IdentityServiceStrategySetup(response.locals.customerId).agent.resolveDid(request.body.issuerDid);
+		if (!resolvedResult?.didDocument || resolvedResult.didDocumentMetadata.deactivated) {
+			return response.status(StatusCodes.BAD_REQUEST).send({
+				error: `${request.body.issuerDid} is either Deactivated or Not found`,
```

Comment:
I agree with @anikitinDSR below actually. Collapsing the `deactivated` and `not found` conditions results in a less helpful message to the user (which case was it?). I would split these to:

1. DID _does_ exist, but it's deactivated. Don't allow issuing a credential in this scenario, no overrides. This kind of check might be better to carry out in did-provider-cheqd. Does Veramo not automatically do this behaviour, i.e., does the built-in `issue` method skip checking whether a DID is deactivated or not? @Eengineer1 
2. A separate condition to test whether a DID was not found. Separating this allow for more helpful error messages. This is available upstream because `resolveDid` has `notFound` property for `DIDResolutionMetadata`

Reply:
Done second request! ✅